data-protocols/ids/ids-api-multipart-endpoint-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/handler/ContractOfferHandler.java

```diff
@@ -82,7 +87,7 @@ public boolean canHandle(@NotNull MultipartRequest multipartRequest) {
         // TODO similar implementation to ContractRequestHandler (only required if counter offers supported, not needed for M1)
 
         return MultipartResponse.Builder.newInstance()
-                .header(ResponseMessageUtil.createRequestInProcessMessage(connectorId, message))
+                .header(responseMessageFactory.createRequestInProcessMessage(message))
```

Comment:
Thanks for pointing this out. I was so focuses on updating the message content, that I forgot to handle the new exceptions in the existing classes accordingly.

The exception should now be added. I also change the type of the `InvalidCorrelationMessageException` to `IllegalArgumentException`, so that its mapped to a 400 status code in the  `EdcApiExceptionMapper`. I assume this will become necessary when the code in a catch-clause is not able to create a rejection message, too.

Reply:
Since both handlers now have the same catch clauses, maybe this could be refactored in another PR to avoid the duplication (assuming more handlers may follow). This PR could also add a method to the ResponseMessageFactory that creates an "empty" `RejectionMessage` with just the rejection reason `INTERNAL_RECIPIENT_ERROR`, that can be returned in case constructing the correct `RejectionMessages` fails, so that the EDC still returns status code 200. (I'm also currently working on a PR to introduce an API context for the IDS endpoints, which will **not** include the `EdcApiExceptionMapper`.)